puzzles/templates/puzzles_table.html

```diff
@@ -1,150 +1,174 @@
+                                          <i class="fa fa-users" aria-hidden="true"></i>
+                                        </span>`;
+                        }
+                        data += `
+    <button type="button" class="editpuzzle btn btn-default" data-toggle="modal" data-target="#editpuzzle" data-pk="${row[pk]}" data-name="${row[name]}" data-url="${row[url]}" data-is_meta="${row[is_meta]}" style="padding: 0px 0px;">
+        <i class="fa fa-edit" aria-hidden="true"></i>
+    </button>
+    <button type="button" class="deletepuzzle btn btn-default" data-toggle="modal" data-target="#deletepuzzle" data-pk="${row[pk]}" data-name="${row[name]}" style="padding: 0px 0px;">
+        <i class="fa fa-trash-o" aria-hidden="true"></i>
+    </button>`;
+                    }
+
+                    return data;
+                },
+            },
+            {
+                'targets': answer,
+                'render': function(data, type, row, meta) {
+                    if (data) {
+                        return data;
+                    }
+                    result = `
+                    <button type="button" class="submitanswer btn btn-outline-primary btn-sm" data-toggle="modal" data-target="#submitanswer" data-pk="${row[pk]}" data-name="${row[name]}">
+                        Submit Answer
+                    </button>`;
+                    return result;
+                }
+            },
+            {
+                'targets': status,
+                'render': function(data, type, row, meta) {
+                    result = row[status];
+                    if (row[status] !== 'PENDING' && row[status] !== 'SOLVED') {
+                        result += `
+<button type="button" class="editstatus btn btn-default" data-toggle="modal" data-target="#editstatus" data-pk="${row[pk]}" data-name="${row[name]}" data-status="${row[status]}" style="padding: 0px 0px;">
+    <i class="fa fa-edit" aria-hidden="true"></i>
+</button>`
+                    }
+                    return result;
+                }
+            },
+            {
+                'targets': sheet,
+                'render': function(data, type, row, meta) {
+                    result = '';
+                    if (data) {
+                        result = `<a href="${data}" target="_blank">Link</a>`;
+                    }
+                    return result;
+                }
+            },
+            {
+                'targets': channel,
+                'render': function(data, type, row, meta) {
+                    result = '';
+                    if (data) {
+                        result = `<a href="${slack_base_url}/app_redirect?channel=${data}" target="_blank">Link</a>`
+                    }
+                    return result;
+                }
+            },
+            {
+                'targets': tags,
+                'render': function(data, type, row, meta) {
+                    result = '';
```

Comment:
nit: var result here and in all the other render functions?

Reply:
Updated. Good to avoid globals.